runtime/compiler/env/j9methodServer.cpp

```diff
@@ -1185,6 +1245,11 @@ TR_ResolvedJ9JITaaSServerMethod::createResolvedJ9MethodMirror(TR_ResolvedJ9JITaa
    std::string methodInfoStr = bodyInfo ? std::string((char*)bodyInfo->getMethodInfo(), sizeof(TR_PersistentMethodInfo)) : std::string();
    std::get<1>(methodInfo) = jbi;
    std::get<2>(methodInfo) = methodInfoStr;
+
+   // set IP method data string.
+   // fanin info is not used at cold opt level, so there is no point sending this information to the server
+   TR_JITaaSClientIProfiler *iProfiler = (TR_JITaaSClientIProfiler *)((TR_J9VMBase *) fe)->getIProfiler();
+   std::get<3>(methodInfo) = (TR::comp() && TR::comp()->getOptLevel() >= warm && iProfiler) ? iProfiler->serializeIProfilerMethodEntry(method) : std::string();
```

Comment:
Is it ever possible for TR::comp() to return 0 at this point?

Reply:
Yes, at the beginning of compilation, a resolved method for the method being compiled gets created, at that point we don't have a comp object yet.